drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,590 @@
+	[FSP_ERR_INVALID_ARGUMENT] = "[FSP_ERR_INVALID_ARGUMENT] Duty cycle is larger than period",
+	[FSP_ERR_UNSUPPORTED] = "[FSP_ERR_UNSUPPORTED] PLL GPT_CFG_OUTPUT_SUPPORT_ENABLE is 0"};
+
+#define ZEPHYR_ERROR_LOG(err)                                                                      \
+	{                                                                                          \
+		if (err == FSP_SUCCESS) {                                                          \
+			(void)0;                                                                   \
+		} else {                                                                           \
+			LOG_ERR("%s: %s", __func__, ra8_pwm_errmsg[err]);                          \
+		}                                                                                  \
+	}
+struct pwm_ra8_capture_data {
+	pwm_capture_callback_handler_t callback;
+	void *user_data;
+	uint64_t period;
+	uint64_t pulse;
+	bool is_pulse_capture;
+	bool is_busy;
+	uint32_t overflows;
+	bool continuous;
+};
+
+struct pwm_ra8_data {
+	gpt_instance_ctrl_t fsp_ctrl;
+	timer_cfg_t fsp_cfg;
+	gpt_extended_cfg_t extend_cfg;
+	uint16_t capture_a_event;
+	uint16_t overflow_event;
+
+#ifdef CONFIG_PWM_CAPTURE
+	struct pwm_ra8_capture_data capture;
+#endif /* CONFIG_PWM_CAPTURE */
+};
+
+struct pwm_ra8_config {
+	const struct device *clock_dev;
+	struct clock_control_ra_subsys_cfg clock_subsys;
+	const struct pinctrl_dev_config *pincfg;
+};
+
+static uint32_t pwm_ra8_gtior_calculate(timer_cfg_t const *const p_cfg,
+					gpt_pin_level_t const stop_level)
+{
+	/* The stop level is used as both the initial level and the stop level. */
+	uint32_t gtior = R_GPT0_GTIOR_OAE_Msk | ((uint32_t)stop_level << R_GPT0_GTIOR_OADFLT_Pos) |
+			 ((uint32_t)stop_level << GPT_PRV_GTIOR_INITIAL_LEVEL_BIT);
+
+	uint32_t gtion = GPT_PRV_GTIO_LOW_COMPARE_MATCH_HIGH_CYCLE_END;
+
+	if (TIMER_MODE_PWM == p_cfg->mode) {
+		/* Use default: GTIOn is high at cycle end, then low at compare match. */
+	}
+
+#if GPT_PRV_EXTRA_FEATURES_ENABLED == GPT_CFG_OUTPUT_SUPPORT_ENABLE
+	else if (p_cfg->mode >= TIMER_MODE_TRIANGLE_WAVE_SYMMETRIC_PWM) {
+		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
+	}
+#endif
+	else if (TIMER_MODE_ONE_SHOT_PULSE == p_cfg->mode) {
+		gtion = GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH;
+	} else {
+		/* In one-shot mode, the output pin goes high after the first compare match (one
+		 * cycle after the timer starts counting).
+		 */
+		if (GPT_PIN_LEVEL_LOW == stop_level) {
+			gtion = GPT_PRV_GTIO_HIGH_COMPARE_MATCH_LOW_CYCLE_END;
+		}
+	}
+
+	gtior |= gtion;
+
+	return gtior;
+}
+
+static int pwm_ra8_apply_gtior_config(gpt_instance_ctrl_t *const p_ctrl,
+				      timer_cfg_t const *const p_cfg)
+{
+	gpt_extended_cfg_t *p_extend = (gpt_extended_cfg_t *)p_cfg->p_extend;
+	uint32_t gtior = p_extend->gtior_setting.gtior;
+
+#if GPT_CFG_OUTPUT_SUPPORT_ENABLE
+
+	/* Check if custom GTIOR settings are provided. */
+	if (0 == p_extend->gtior_setting.gtior) {
```

Comment:
```suggestion
	if (p_extend->gtior_setting.gtior == 0) {
```

I think I know what's behind the inverted condition but that's very unusual in the Zephyr code base and just makes the codde look unfamiliar.

Same for the others below.

Reply:
I updated it, thanks